src/time_integration/paired_explicit_runge_kutta/methods_PERK3.jl

```diff
@@ -0,0 +1,366 @@
+end
+
+# Constructor that calculates the coefficients with polynomial optimizer from a list of eigenvalues
+function PairedExplicitRK3(num_stages, tspan, eig_vals::Vector{ComplexF64};
+                           verbose = false, cS2 = 1.0f0)
+    a_matrix, c = compute_PairedExplicitRK3_butcher_tableau(num_stages,
+                                                            tspan,
+                                                            eig_vals;
+                                                            verbose, cS2)
+    return PairedExplicitRK3(num_stages, a_matrix, c)
+end
+
+# This struct is needed to fake https://github.com/SciML/OrdinaryDiffEq.jl/blob/0c2048a502101647ac35faabd80da8a5645beac7/src/integrators/type.jl#L77
+# This implements the interface components described at
+# https://diffeq.sciml.ai/v6.8/basics/integrator/#Handing-Integrators-1
+# which are used in Trixi.
+mutable struct PairedExplicitRK3Integrator{RealT <: Real, uType, Params, Sol, F, Alg,
+                                           PairedExplicitRKOptions} <:
+               AbstractPairedExplicitRKSingleIntegrator
+    u::uType
+    du::uType
+    u_tmp::uType
+    t::RealT
+    tdir::RealT
+    dt::RealT # current time step
+    dtcache::RealT # manually set time step
+    iter::Int # current number of time steps (iteration)
+    p::Params # will be the semidiscretization from Trixi
+    sol::Sol # faked
+    f::F
+    alg::Alg # This is our own class written above; Abbreviation for ALGorithm
+    opts::PairedExplicitRKOptions
+    finalstep::Bool # added for convenience
+    dtchangeable::Bool
+    force_stepfail::Bool
+    # PairedExplicitRK stages:
+    k1::uType
+    k_higher::uType
+end
+
+function init(ode::ODEProblem, alg::PairedExplicitRK3;
+              dt, callback = nothing, kwargs...)
+    u0 = copy(ode.u0)
+    du = zero(u0)
+    u_tmp = zero(u0)
+
+    # PairedExplicitRK stages
+    k1 = zero(u0)
+    k_higher = zero(u0)
+
+    t0 = first(ode.tspan)
+    tdir = sign(ode.tspan[end] - ode.tspan[1])
+    iter = 0
+
+    integrator = PairedExplicitRK3Integrator(u0, du, u_tmp, t0, tdir, dt, dt, iter,
+                                             ode.p,
+                                             (prob = ode,), ode.f, alg,
+                                             PairedExplicitRKOptions(callback,
+                                                                     ode.tspan;
+                                                                     kwargs...),
+                                             false, true, false,
+                                             k1, k_higher)
+
+    # initialize callbacks
+    if callback isa CallbackSet
+        for cb in callback.continuous_callbacks
+            error("unsupported")
+        end
+        for cb in callback.discrete_callbacks
+            cb.initialize(cb, integrator.u, integrator.t, integrator)
+        end
+    elseif !isnothing(callback)
+        error("unsupported")
```

Comment:
Could this also be done by specifying this in the function signature `callback::Union{CallbackSet, Nothing} = nothing` (see above)?

Reply:
This works and IMHO is a more compact way of dealing with the scenario. However, like other issues you pointed out with this integrator, this is a common convention across all integrators. I think a new PR that addresses all integrators might be a better place to tackle this. (I can open this)

Reply:
Yes, I agree. I'm fine with that.

Reply:
See above.